resources/files/windows/PuppetBolt/PuppetBoltFunctions.ps1

```diff
@@ -0,0 +1,1639 @@
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ConfigFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $InventoryFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $SaveRerun,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Transport,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $ConnectTimeout,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $TTY,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Tmpdir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $Color,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [BoltOutputLevel]  $OutputLevel,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $TargetCommand,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $PlanName,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $TaskName,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $BoltCommand,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ScriptLocation,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $PuppetManifest,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Source,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Destination,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Targets,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Query
+    )
+    begin {
+        $_flags = Convert-BoltParams @PSBoundParameters
+    }
+    process {
+        if ($Targets) {
+            $_targets = "--nodes=$Targets"
+        } elseif ($Query) {
+            $_targets = "--query=$Query"
+        } else {
+            Write-Error "Must specify one of -Targets,-Query"
+            throw
+        }
+        if ($OutputLevel) {
+            $_result = Invoke-Expression "bolt $CommandString $_targets --format=json $_flags"
+        } else {
+            $_result = Invoke-Expression "bolt $CommandString $_targets --format=json $_flags" 2>$null
+        }
+        return ($_result | ConvertFrom-Json)
+    }
+}
+
+function Invoke-BoltTask {
+    <#
+    .SYNOPSIS
+    Execute Puppet Bolt task
+    .DESCRIPTION
+    This function will execute 'bolt task run <task>' on the targets specified by the -Targets
```

Comment:
Not sure if it's useful to say what the underlying bolt command line is.  Will a PowerShell user care?